core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -73,6 +73,7 @@ public class CairoEngine implements Closeable, WriterSource {
     protected final CairoConfiguration configuration;
     private final AtomicLong asyncCommandCorrelationId = new AtomicLong();
     private final CopyContext copyContext;
+    private final ConcurrentHashMap<TableToken> createTableLock = new ConcurrentHashMap<>();
```

Comment:
We already have a similar map in `TableNameRegistryRW` (`tableNameToTableTokenMap`). So, maybe we could add `boolean lockTableName(TableToken)` method to `TableNameRegistry` and use it `O3PartitionPurgeJob#purgePartition()`?

Reply:
I considered it but couldn't find a way to make it work.
TableNameRegistry cannot block an existing table from being re-created. Once it's dropped it is available to be created again. To lock the table from creation an additional data structure is needed.